### PR TITLE
Automatically create new user on (potential) Sandbox reset.

### PR DESCRIPTION
**Pull request by JordyHeemskerk**
Closes  bunq/tinker_php#11

### CODE REVIEW

File: src/BunqLib.php [hunk 1]
```diff
@@ -96,11 +98,21 @@ private function setupContext()
...
-        $apiContext->save($this->determineBunqConfFileName());
-
-        BunqContext::loadApiContext($apiContext);
+        try {
+            $apiContext = ApiContext::restore($this->determineBunqConfFileName());
+            $apiContext->ensureSessionActive();
+            $apiContext->save($this->determineBunqConfFileName());
+            BunqContext::loadApiContext($apiContext);
+        } catch (ForbiddenException $forbiddenException) {
+            if (BunqEnumApiEnvironmentType::SANDBOX()->equals($this->environment)
```
Comment (OGKevin) [inline src/BunqLib.php]: Extract this in its own method for better readability. 

```
private function handleForbiddenException(ForbiddenException $exception)
{
   // the code inside the catch block 
}

```

File: src/BunqLib.php [hunk 2]
```diff
@@ -96,11 +98,21 @@ private function setupContext()
...
+                && strpos($forbiddenException->getMessage(), self::ERROR_INSUFFICIENT_AUTHENTICATION) !== false
```
Comment (OGKevin) [inline src/BunqLib.php]: Hmm we normally use REGEX for these type of lookups instead of `strpos`.

**Review by OGKevin — CHANGES_REQUESTED** [review]
Some minor refactor requests.

File: src/BunqLib.php [hunk 3]
```diff
@@ -143,6 +136,23 @@ private function setupCurrentUser()
...
 
+    /**
+     * @param ForbiddenException $forbiddenException
+     *
+     * @throws ForbiddenException
+     */
+    private function handleForbiddenException(ForbiddenException $forbiddenException)
+    {
+        if (BunqEnumApiEnvironmentType::SANDBOX()->equals($this->environment)
+            && preg_match(self::REGEX_ERROR_INSUFFICIENT_AUTHENTICATION, $forbiddenException->getMessage())
```
Comment (OGKevin) [inline src/BunqLib.php]:
```
preg_match(self::REGEX_ERROR_INSUFFICIENT_AUTHENTICATION, $forbiddenException->getMessage()) ===  self::REGEX_MATCH_FOUD
```

where `REGEX_MATCH_FOUD = 1`

> preg_match() returns 1 if the pattern matches given subject, 0 if it does not, or FALSE if an error occurred.
http://php.net/manual/en/function.preg-match.php

**Review by OGKevin — CHANGES_REQUESTED** [review]
One small change.

File: src/BunqLib.php (hunk 3 above)
Comment (JordyHeemskerk) [reply src/BunqLib.php]: That is not required as:
```
php > if (1) echo "true" . PHP_EOL;
true
php > if (0) echo "true" . PHP_EOL;
php > if (false) echo "true" . PHP_EOL;

Comment (OGKevin) [reply src/BunqLib.php]: Yes, but this is some php magic ✨ and unfortunately we don't use this e.g.

https://github.com/bunq/sdk_php/blob/c0895dc40c7076b103bbb931ec69068e4daf94f0/src/Util/InstallationUtil.php#L282-L289

We dont want to rely on php auto cast.

**Review by OGKevin — APPROVED** [review]
LGTM

**Comment by OGKevin** [comment]
@andrederoos

**Pull request merged**